### DISCUSSION TITLE
Driving motors with motor controllers vs directly off digital pins in miniature controller version

### DISCUSSION THREAD

**Discussion by ptparnell** [pdbuzzboard/pdbuzzboard#10] [Q&A]
First, I want to say Thank you to the pdbuzzboard developer who has put so much time and effort in sharing his work with everyone with the aim of helping other people.  My wife also has PD and wants me to make her a glove version.  The first controller for the buzz board used an Arduino UNO R3 with a quad L293 motor controller shield but the miniaturized version used with the glove application uses an Arduino Nano that drives the motors directly from digital output pins.  Why doesn't the second incarnation require motor controller circuitry?  Do the digital pins provide sufficient current to drive the motors the same as the motor controller circuits?  Do the digital pins have built in protection that protect them from motor noise?  It was quite a leap.  Thank you

**Comment by pdbuzzboard** [comment]
The micro controller version delivers slightly less voltage than the motor shield, but sufficient to get a good vibration measurements. There is no motor noise protection at all. My first incarnation of the microcontrollerused PNPs and resistors to isolate the power flow from the signal.  The current iteration was driving toward making the controller as small as possible.On Jun 20, 2023, at 9:43 PM, ptparnell ***@***.***> wrote:﻿
First, I want to say Thank you to the pdbuzzboard developer who has put so much time and effort in sharing his work with everyone with the aim of helping other people.  My wife also has PD and wants me to make her a glove version.  The first controller for the buzz board used an Arduino UNO R3 with a quad L293 motor controller shield but the miniaturized version used with the glove application uses an Arduino Nano that drives the motors directly from digital output pins.  Why doesn't the second incarnation require motor controller circuitry?  Do the digital pins provide sufficient current to drive the motors the same as the motor controller circuits?  Do the digital pins have built in protection that protect them from motor noise?  It was quite a leap.  Thank you

**Reply by ptparnell** [reply]
The cheaper motors typically run at 3Vdc at 85ma.  The Vybronics typically run at 67ma (90ma max) at 3Vdc.  The I/O pins of the Nano driven by the ATmega328 typically will provide 20ma @ 4.5V (VCC = 5V  @ 25C).  The typical maximum current the pins can provide without damage is 40ma according to the datasheet.  This would imply that though the motors are running they are probably not running at full speed if driven directly from the Nano without a motor driver module.  Also note the Vybronics motor which is available at digi-key and states 15000RPM, 3VDC has a very wide range of speed 15000 +/- 3000 RPM.  So if people are using the nano design and getting a positive result it would seem the motor speed is not that critical.  Food for thought.

**Reply by ptparnell** [reply]
The L293d Arduino motor drivers on the shield are powered by 5V from the external supply.  The Vybronics DC motors operating voltage is 2.7Vdc to 3.3Vdc.  There is no maximum voltage listed in the datasheet.  I wonder if the 5V supply is stressing the motors?

**Comment by pdbuzzboard** [comment]
The motor shield has speed (voltage) control and backs down the 5v by about 20%On Jun 23, 2023, at 1:09 AM, ptparnell ***@***.***> wrote:﻿
The L293d Arduino motor drivers on the shield are powered by 5V from the external supply.  The Vybronics DC motors operating voltage is 2.7Vdc to 3.3Vdc.  There is no maximum voltage listed in the datasheet.  I wonder if the 5V supply is stressing the motors?

**Comment by Domminz** [comment]
Hi Everyone,
As i understand, rotary vibration motors are being used. If one wishes to use the Linear Vibration motor (AC) what iterations are required in Hardware & Software. The motor specifications are as under:
Product: Linear Coin Vibration Motor
Operating Voltage: 0.1~1.9 VrmsAC
Acceleration: G 1.7
Direction of vibration: Z axis
Rated Current: 80mA Max

**Reply by pdbuzzboard** [reply]
LRA motors require very different circuitry and code.  ERMs work based upon supplying and removing voltage (DC) to make them spin.  LRA require AC current at the desired frequency.  Both Sparkfun and Adafruit have LRA driver boards.  As the LRA boards are all shipped with the same address, you will need a multiplexer to switch between designated motors in order to deliver the desired pattern.  An individual put together the code to drive the Adafruit design and put it on github here: github.com/kriswilk/buzzah .  I have assembled it and it does vibrate and follow pattern.  I haven't had the time to take it, or the Sparkfun, further.  Sparkfun has a solution that may offer a "no solder" LRA solution (better suited for a board than a glove).
Please share if you pursue.
